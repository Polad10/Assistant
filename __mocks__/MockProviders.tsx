import { Dispatch, PropsWithChildren, SetStateAction } from 'react'
import { Translator } from '../helpers/Translator'
import { AppointmentRequest } from '../modals/Appointment'
import { PatientRequest } from '../modals/Patient'
import { Setting } from '../modals/Setting'
import { TreatmentRequest } from '../modals/Treatment'
import { PaymentRequest } from '../modals/Payment'
import { Api } from '../helpers/Api'
import { mockModels } from './MockModels'
import { AuthContext } from '../contexts/AuthContext'
import { DataContext } from '../contexts/DataContext'
import { ThemeContext } from '../contexts/ThemeContext'
import { LocalizationContext } from '../contexts/LocalizationContext'

const themeContextMock = {
  primary: '#121212',
  secondary: '#1a1a1a',
  accent: '#7600eb',
  neutral: '#faf8ff',
  border: '#374151',
  info: '#d3fbd8',
  success: '#00e5b0',
  warning: '#9f1239',
  error: '#9f1239',
}

const localizationContextMock = {
  translator: new Translator('en'),
  language: 'en',
}

const authContextMock = {
  user: undefined,
  loading: false,
}

const dataContextMock = {
  fetchPatients: async () => {},
  fetchAppointments: async () => {},
  fetchTreatments: async () => {},
  fetchPayments: async () => {},
  fetchSetting: async () => {},

  createAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  updateAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  deleteAppointment: async (appointmentId: number) => {},

  createPatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  updatePatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  deletePatient: async (patientId: number) => {},

  createTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  updateTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  deleteTreatment: async (treatmentId: number) => {},

  createPayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  updatePayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  deletePayment: async (paymentId: number) => {},

  createSetting: async (setting: Setting) => {},
  updateSetting: async (setting: Setting) => {},

  api: undefined,
  setApi: (action: SetStateAction<Api | undefined>): Dispatch<SetStateAction<Api | undefined>> => jest.fn(),

  loading: false,
  setLoading: (action: SetStateAction<boolean>): Dispatch<SetStateAction<boolean>> => jest.fn(),

  patients: [mockModels.patient_1, mockModels.patient_2],
  appointments: [mockModels.appointment_1, mockModels.appointment_2],
  treatments: [mockModels.treatment_1, mockModels.treatment_2],
  payments: [mockModels.payment_1, mockModels.payment_2],
  setting: mockModels.setting,
}

export default function MockProviders({ children }: PropsWithChildren) {
  return (
    <AuthContext.Provider value={authContextMock}>
      <DataContext.Provider value={dataContextMock}>
        <ThemeContext.Provider value={themeContextMock}>
          <LocalizationContext.Provider value={localizationContextMock}>{children}</LocalizationContext.Provider>
        </ThemeContext.Provider>
      </DataContext.Provider>
    </AuthContext.Provider>
  )
}
