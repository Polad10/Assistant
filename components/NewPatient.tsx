import { useCallback, useContext, useEffect, useState } from 'react'
import PatientForm from './PatientForm'
import { DeviceEventEmitter } from 'react-native'
import { PatientRequest } from '../modals/Patient'
import { DataContext } from '../contexts/DataContext'
import { useNavigation } from '@react-navigation/native'
import { RootStackScreenProps } from '../types/Navigation'
import { showSuccessMessage } from '../helpers/ToastHelper'
import MainView from './MainView'
import LoadingView from './LoadingView'

export default function NewPatient() {
  const navigation = useNavigation<RootStackScreenProps<'NewPatient'>['navigation']>()
  const context = useContext(DataContext)!

  const [loading, setLoading] = useState(false)

  const handlePatientSave = useCallback(async (patient: PatientRequest) => {
    setLoading(true)
    const newPatient = await context.createPatient(patient)
    setLoading(false)

    showSuccessMessage('Patient added')

    if (DeviceEventEmitter.listenerCount('patientCreated') > 0) {
      DeviceEventEmitter.emit('patientCreated', newPatient)
    } else {
      navigation.goBack()
    }
  }, [])

  useEffect(() => {
    const listener = DeviceEventEmitter.addListener('patientSaved', handlePatientSave)

    return () => {
      listener.remove()
    }
  }, [])

  return (
    <MainView>
      <PatientForm pageName='NewPatient' />
      {loading && <LoadingView />}
    </MainView>
  )
}
