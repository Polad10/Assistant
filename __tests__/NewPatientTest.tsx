import renderer from 'react-test-renderer'
import MockProviders from '../__mocks__/MockProviders'
import NewPatient from '../components/NewPatient'

describe('NewPatient', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MockProviders>
          <NewPatient />
        </MockProviders>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
