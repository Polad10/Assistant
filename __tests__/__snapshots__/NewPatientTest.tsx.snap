// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPatient renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#121212",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "backgroundColor": "#121212",
          "flex": 1,
        },
        {
          "paddingHorizontal": 10,
          "paddingTop": 20,
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        [
          {
            "flex": 1,
          },
          {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          {
            "flexGrow": 1,
          }
        }
        onScroll={[Function]}
        scrollEventThrottle={50}
      >
        <View>
          <View
            style={
              {
                "paddingHorizontal": 0,
                "width": "100%",
              }
            }
            testID="RNE__Input__view-wrapper"
          >
            <Text
              style={
                {
                  "color": "#d3fbd8",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              First Name
            </Text>
            <View
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "borderBottomWidth": 1,
                  "borderColor": "#374151",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "marginBottom": 20,
                  "transform": [
                    {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <TextInput
                autoCorrect={false}
                editable={true}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Egana"
                placeholderTextColor="#86939e"
                returnKeyType="done"
                selectionColor="#faf8ff"
                showError={false}
                style={
                  {
                    "color": "#faf8ff",
                    "flex": 1,
                    "fontSize": 18,
                    "minHeight": 40,
                    "paddingHorizontal": 10,
                    "paddingTop": 20,
                    "paddingVertical": 20,
                  }
                }
                testID="RNE__Input__text-input"
                underlineColorAndroid="transparent"
              />
            </View>
            <Text
              style={
                {
                  "color": "#ff190c",
                  "fontSize": 12,
                  "height": 0,
                  "margin": 0,
                  "padding": 0,
                }
              }
            />
          </View>
          <View
            style={
              {
                "paddingHorizontal": 0,
                "width": "100%",
              }
            }
            testID="RNE__Input__view-wrapper"
          >
            <Text
              style={
                {
                  "color": "#d3fbd8",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              Last Name
            </Text>
            <View
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "borderBottomWidth": 1,
                  "borderColor": "#374151",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "marginBottom": 20,
                  "transform": [
                    {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <TextInput
                autoCorrect={false}
                editable={true}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Mammadova"
                placeholderTextColor="#86939e"
                returnKeyType="done"
                selectionColor="#faf8ff"
                showError={false}
                style={
                  {
                    "color": "#faf8ff",
                    "flex": 1,
                    "fontSize": 18,
                    "minHeight": 40,
                    "paddingHorizontal": 10,
                    "paddingTop": 20,
                    "paddingVertical": 20,
                  }
                }
                testID="RNE__Input__text-input"
                underlineColorAndroid="transparent"
              />
            </View>
            <Text
              style={
                {
                  "color": "#ff190c",
                  "fontSize": 12,
                  "height": 0,
                  "margin": 0,
                  "padding": 0,
                }
              }
            />
          </View>
          <View>
            <View
              style={
                {
                  "paddingHorizontal": 0,
                  "width": "100%",
                }
              }
              testID="RNE__Input__view-wrapper"
            >
              <Text
                style={
                  {
                    "color": "#d3fbd8",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginBottom": 5,
                  }
                }
              >
                Date of birth
              </Text>
              <View
                collapsable={false}
                style={
                  {
                    "alignItems": "center",
                    "borderBottomWidth": 1,
                    "borderColor": "#374151",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "marginBottom": 20,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <TextInput
                  autoCorrect={false}
                  editable={true}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="2014-01-01"
                  placeholderTextColor="#86939e"
                  returnKeyType="done"
                  selectionColor="#faf8ff"
                  style={
                    {
                      "color": "#faf8ff",
                      "flex": 1,
                      "fontSize": 18,
                      "minHeight": 40,
                      "paddingHorizontal": 10,
                      "paddingTop": 20,
                      "paddingVertical": 20,
                    }
                  }
                  testID="RNE__Input__text-input"
                  underlineColorAndroid="transparent"
                />
                <View
                  style={
                    {
                      "alignItems": "center",
                      "height": 40,
                      "justifyContent": "center",
                      "marginRight": 10,
                      "marginVertical": 4,
                      "paddingRight": 0,
                    }
                  }
                >
                  <Text />
                </View>
              </View>
              <Text
                style={
                  {
                    "color": "#ff190c",
                    "fontSize": 12,
                    "height": 0,
                    "margin": 0,
                    "padding": 0,
                  }
                }
              />
            </View>
            <Modal
              animationType="none"
              hardwareAccelerated={false}
              isVisible={false}
              onHide={[Function]}
              transparent={true}
              visible={false}
            />
          </View>
          <View
            style={
              {
                "paddingHorizontal": 0,
                "width": "100%",
              }
            }
            testID="RNE__Input__view-wrapper"
          >
            <Text
              style={
                {
                  "color": "#d3fbd8",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              City
            </Text>
            <View
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "borderBottomWidth": 1,
                  "borderColor": "#374151",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "marginBottom": 20,
                  "transform": [
                    {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <TextInput
                autoCorrect={false}
                editable={true}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Baku"
                placeholderTextColor="#86939e"
                returnKeyType="done"
                selectionColor="#faf8ff"
                style={
                  {
                    "color": "#faf8ff",
                    "flex": 1,
                    "fontSize": 18,
                    "minHeight": 40,
                    "paddingHorizontal": 10,
                    "paddingTop": 20,
                    "paddingVertical": 20,
                  }
                }
                testID="RNE__Input__text-input"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <Text
              style={
                {
                  "color": "#ff190c",
                  "fontSize": 12,
                  "height": 0,
                  "margin": 0,
                  "padding": 0,
                }
              }
            />
          </View>
          <View
            style={
              {
                "paddingHorizontal": 0,
                "width": "100%",
              }
            }
            testID="RNE__Input__view-wrapper"
          >
            <Text
              style={
                {
                  "color": "#d3fbd8",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              Phone Number
            </Text>
            <View
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "borderBottomWidth": 1,
                  "borderColor": "#374151",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "marginBottom": 20,
                  "transform": [
                    {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <TextInput
                autoCorrect={false}
                editable={true}
                keyboardType="phone-pad"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="050 123 45 67"
                placeholderTextColor="#86939e"
                returnKeyType="done"
                selectionColor="#faf8ff"
                style={
                  {
                    "color": "#faf8ff",
                    "flex": 1,
                    "fontSize": 18,
                    "minHeight": 40,
                    "paddingHorizontal": 10,
                    "paddingTop": 20,
                    "paddingVertical": 20,
                  }
                }
                testID="RNE__Input__text-input"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <Text
              style={
                {
                  "color": "#ff190c",
                  "fontSize": 12,
                  "height": 0,
                  "margin": 0,
                  "padding": 0,
                }
              }
            />
          </View>
          <View
            style={
              {
                "paddingHorizontal": 0,
                "width": "100%",
              }
            }
            testID="RNE__Input__view-wrapper"
          >
            <Text
              style={
                {
                  "color": "#d3fbd8",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              Extra Info
            </Text>
            <View
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "borderBottomWidth": 1,
                  "borderColor": "#374151",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "marginBottom": 20,
                  "transform": [
                    {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <TextInput
                autoCorrect={false}
                editable={true}
                multiline={true}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="A friend of Polad."
                placeholderTextColor="#86939e"
                returnKeyType="default"
                selectionColor="#faf8ff"
                style={
                  {
                    "color": "#faf8ff",
                    "flex": 1,
                    "fontSize": 18,
                    "minHeight": 100,
                    "paddingHorizontal": 10,
                    "paddingTop": 15,
                    "paddingVertical": 20,
                  }
                }
                testID="RNE__Input__text-input"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <Text
              style={
                {
                  "color": "#ff190c",
                  "fontSize": 12,
                  "height": 0,
                  "margin": 0,
                  "padding": 0,
                }
              }
            />
          </View>
        </View>
      </RCTScrollView>
    </View>
    <RCTSafeAreaView
      style={
        {
          "marginTop": "auto",
        }
      }
    >
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 2,
            },
            undefined,
            false,
          ]
        }
        testID="RNE_BUTTON_WRAPPER"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": false,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "marginHorizontal": 10,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#7600eb",
                "borderColor": "#2089dc",
                "borderRadius": 10,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 10,
                "paddingHorizontal": 10,
              }
            }
          >
            <Text
              style={
                {
                  "color": "white",
                  "fontSize": 18,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              Create
            </Text>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</View>
`;
